### PR TITLE
finished coding challenge

### PR DESCRIPTION
**Pull request by JBajracharya**
- [ x] Top-level README “Table of Contents” is updated
 - [ x] Feature tasks for this challenge are completed
 - [x ] Unit tests written and passing
     - [x ] “Happy Path” - Expected outcome
     - [ x] Expected failure
     - [x ] Edge Case (if applicable/obvious)
 - [ x] README for this challenge is complete
     - [ x] Summary, Description, Approach & Efficiency, Solution
     - [ x] Link to code
     - [x ] Picture of whiteboard

### CODE REVIEW

File: code401challenges/src/main/java/code401challenges/BinarySearch.java [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+    public static int binarySearch(int[] sortedArray, int searchKey) {
+        int firstIndex = 0;
+        int lastIndex = sortedArray.length - 1;
+        while (firstIndex <= lastIndex) {
+            int midpointIndex = (int)Math.ceil((firstIndex + lastIndex)/2);
+            if(searchKey < sortedArray[midpointIndex]) {
+                lastIndex = midpointIndex - 1;
+            } else if(searchKey > sortedArray[midpointIndex]) {
+                firstIndex = midpointIndex + 1;
+            } else return midpointIndex;
```
Comment (mnfmnfm) [inline code401challenges/src/main/java/code401challenges/BinarySearch.java]: Tiny nitpick: I always like to use curly braces, even for single-line conditionals, to avoid the [goto fail](https://www.wired.com/2014/02/gotofail/).

File: code401challenges/src/test/java/code401challenges/BinarySearchTest.java [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+import static org.junit.Assert.assertEquals;
+
+public class BinarySearchTest {
+    @Test public void testBinarySearch() {
+        int[] inputArray = {4, 8, 26, 15, 43, 18, 10, 12};
+        int inputNumber = 15;
+        int output  = 3;
+        int actual = BinarySearch.binarySearch(inputArray, inputNumber);
+
+        assertEquals("Expected 15", output, actual);
```
Comment (mnfmnfm) [inline code401challenges/src/test/java/code401challenges/BinarySearchTest.java]: A good single test to start with! Note that the testing requirements say you should, at minimum, test for a happy path and an expected failure (like a missing number).
